ublox/src/ubx_packets/packets.rs

```diff
@@ -1970,15 +1973,55 @@ struct RxmRtcm {
 }
 
 #[ubx_packet_recv]
-#[ubx(class = 0x10, id = 0x02, fixed_payload_len = 16)]
+#[ubx(class = 0x10, id = 0x02, max_payload_len = 1240)]
 struct EsfMeas {
     time_tag: u32,
+    #[ubx(map_type = EsfMeasInfo, from = EsfMeasInfo::new)]
+    info: [u8; 0],
+}
+
+pub struct EsfMeasInfo<'a> {
     flags: u16,
     id: u16,
-    /// flags >> 11 & 0x1F
-    data: u32,
-    /// flags & 0x8
-    calib_tag: u32,
+    data: Box<dyn Iterator<Item = &'a u32>>,
+    calib_tag: Option<u32>,
+}
+
+impl<'a> EsfMeasInfo<'a> {
+    fn new(bytes: &'a [u8]) -> Self {
+        let mut chunks = bytes.chunks_exact(2);
+        let flags = u16::from_le_bytes(chunks.next().unwrap().try_into().unwrap());
+        let id = u16::from_le_bytes(chunks.next().unwrap().try_into().unwrap());
+
+        let calib_tag = if flags & 0x8 != 0 {
+            let first_last: [u8; 2] = chunks.next_back().unwrap().try_into().unwrap();
+            let second_last: [u8; 2] = chunks.next_back().unwrap().try_into().unwrap();
+            Some(u32::from_le_bytes(
```

Comment:
apparently tupling it up will remove the `DoubleEnded` iter so will have to manually form the tuple beforehand and construct the u32...